Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -17,20 +15,7 @@ public GildedRose(Item[] items) {
 
     public void updateQuality() {
         for (Item item : items) {
-            itemFactory.customiseItem(item).updateState();
-            if (hasReachedLowestQualityValue(item)) {
-                item.quality = QualityValues.lowestValuePossible();
-            } else if (hasReachedHighestQualityValue(item)) {
-                item.quality = QualityValues.highestValuePossible(item);
-            }
+            itemFactory.customiseItem(item).updateState(item);
```

Comment:
👍  Good that you managed to remove the if statement below

Reply:
Why do we need to pass the item twice? How about the `CustomisedItem` remembers what item it wraps?